### DISCUSSION TITLE
Ainda vale a pena aprender JQuery?

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#1080] [Discussão]
Na opinião de vcs vale a pena aprender JQuery? ele ainda esta sendo utilizado ou já ficou para trás?

**Comment by bizarro** [comment]
Sou daqueles que **odeia** usar jQuery nos dias atuais com todas as forças. :joy: Pode até ser que ajude muita gente em algumas tarefas, mas hoje em dia as coisas estão bem padronizadas entre os diversos dispositivos e navegadores que não existe motivo algum para usar. Para mim ficou uma coisa automática como por exemplo.

Quer usar um elemento da página? `querySelector()`. Quer adicionar uma classe em algum elemento? `classList.add()`. Quer remover essa classe? `classList.remove()`. Quer adicionar um atributo? `setAttribute('src', 'valor')`. Quer fazer uma requisição AJAX? Use `XMLHttpRequest()` ou `window.fetch` com pollyfill. Quer adicionar um event listener num elemento? `window.addEventListener('visibilitychange', evento)`. E por aí vai, é mais questão de costume mesmo, o que me ajudou lá no começo quando eu não manjava muito foi o [You DON'T Need jQuery](https://github.com/nefe/You-Dont-Need-jQuery).

Agora se você vai trabalhar com uma Single Page Application em Angular, React, Vue ou qualquer outro framework, *nunca use jQuery com eles*, até porque eles já tem todo o toolkit necessário para manipular o DOM, fazer requisições e etc. :sweat_smile:

**Comment by didiraja** [comment]
Como iniciante eu digo que vale pra aprender e se acostumar com javascript. Porque voce consegue com facilidade fazer coisas e é até mais fácil de achar soluções para problemas na internet. 

**PORÉM,** sabendo que é uma fase temporária de iniciante. Porque a comunidade ta convergindo para a não-necessidade de jquery e uso do JS puro, o que é melhor. Sem contar, que realmente acredito na aposentadoria do jQuery. Questão de tempo.

**Comment by ninetails** [comment]
Acho válido somente em dois casos: total iniciante e "programadores" de WordPress.

Em ambos os casos, é válido aprender pq é o código que vai estar por aí no Google pra ser copiado :rofl:

<sub><sup>**Edit:**  era pra ser uma piada</sup></sub>

**Comment by eduardojmatos** [comment]
Eu acho válido se você estiver num cenário que precise dar suporte a browsers IE9 > e alguns mobile mais antigos. 

E dependendo do que você for fazer também, porque se for por conta do `ajax`, tem libs menores que fazem isso pra você. 

Se for manipulação de DOM (eventos, add/remove, etc.), ainda vale usar, na minha opinião, se precisar desse suporte pra browsers antigos.

**Comment by felipefialho** [comment]
Como o @eduardojmatos disse, se precisar dar suporte para browsers antigos, usa sem dó. 

Ou mesmo, em um projeto estático, especialmente em sites institucionais e landing pages, se achar que o jQuery vai contribuir na velocidade de desenvolvimento, talvez vale a pena, especialmente se carregar direto de uma CDN.

Mas respondendo a pergunta (se vale a pena estudar jQuery)... eu acho que não. Acho jQuery intuitivo pra aprender conforme você precise fazer as coisas dentro dele, melhor focar em JavaScript puro ou mesmo libs mais robustas (React, Angular, Vue)...

**Comment by felipeqburian** [comment]
Poxa, eu sou bem iniciante no JS, aprendi agora a pouco na faculdade o uso, JQuery na minha opinião é interessante, tu não precisa escrever nem decorar alguns comandos extensos, botar algumas manipulações na página é bem prático, achei muito interessante, a otimização do código fica muito boa, tu se perde muito pouco no código, já sabe onde precisa mexer e qual função criar/editar para determinado elemento ficar da forma que você precisa com alguma ação ou evento.

Minha opinião é que este tipo de conhecimento ajuda, eu particularmente estou gostando (não tenho conhecimento dos novos frameworks ainda), mas como a maioria dos desenvolvedores em JS estão migrando para os novos Frameworks, muito mais robustos e bem otimizados, acho também que é questão de tempo para ele virar um coadjuvante, não acho que será obsoleto.

**Comment by AmandaFonseca** [comment]
A ultima vez que usei Jquery foi para usar requisição AJAX e só kkk mas no começo ele quebrou muito o meu galho. Hoje em dia se eu estou com o prazo de entrega estourando tipo aos 45 do segundo tempo e eu encontrei uma solução pro meu problema no stackoverflow que usa  Jquery eu vou lá e pego o código, faço as adaptações para o meu projeto, testo e já entrego.  Ter conhecimento em Jquery sempre é bom porque pode acontecer um caso como esse que falei mas quem tem uma boa base de javascript não precisa. Eu estou aprofundando meus conhecimentos em javascript e agora já testo funções simples de validação direto no console (não sei se isso faz diferença pra você mas eu gosto kk) além de entender melhor o funcionamento de códigos e plugins.

**Comment by bernardodiasc** [comment]
jQuery faz parte da história do desenvolvimento web. Aprender essa lib soh vai te deixar mais inteligente, acho q não tem problema algum nisso.

Agora a questão usar ou não, eh uma pergunta q serve pra qualquer ferramenta... A resposta eh sempre a mesma: depende.

**Comment by gerardo-junior** [comment]
Acho super valido, até mesmo pra compatibilidade, você pode pegar muito codigo legado pra fazer alguma mudaça ou manutenção, jquery ainda tá muito vivo por ae pra ser descartado totalmente.

e pra uso as vezes (as vezes) umas pog ficam menos terrivel quando feita com jquery.

**Comment by phenriqueleao** [comment]
Acho válido, por mais que hoje em dia as coisas estejam se afastando dele, ainda existe muito código legado com ele.

**Comment by cezarsmpio** [comment]
Aprender por aprender? Não.

Precisa dar manutenção em sites legados com jQuery? Se souber como JavaScript funciona, não.

Usar em um projeto novo? Não.

Dar suporte para browsers muito antigos, IE8, 9 e 10? Sim, ajuda.

**Comment by ghost** [comment]
Obrigado pela opinião de vcs galera, vcs são zikas ;)

**Comment by simaodeveloper** [comment]
Se você constrói Website, precisa de suporte Cross-Browser, precisa entregar em um prazo curto: 

# Sim vale a pena!

**Comment by felipelssilva** [comment]
com o jquery você consegue ir longe na área, muitos sistemas funcionam em cima do jquery. Se um algum dia ele for descontinuado, com certeza aparecerá uma nova tecnologia imensamente igual/melhor!!

**Comment by expalmer** [comment]
Aprender jQuery é aprender um pouco da história do JS na web. O ideal é aprender o porque do jQuery existir e principalmente criar o teu próprio "jQuery de pobre" para entender os problemas que ele resolve.

**Comment by simaodeveloper** [comment]
Acrescentando:

Depende do projeto!

Essa pergunta é facilmente respondida lendo a documentação do JQuery,
desse modo você entender os problemas que ele resolve, sabendo isso,
você conseguirá decidir se ele resolve os problemas do seu projeto.

Vou acrescentar alguns pontos:

- JQuery é simples, fácil de ler e entender;
- É CrossBrowser;
- A Web é composta por Websites, e caso precise construir-los na minha opinião o JQuery é a melhor opção para entregar um resultado satisfatório em um prazo pequeno.
- Manipula DOM (Events, Traversing, Forms, Attributes, entre outros), realiza requisições HTTP (AJAX), cria Animações, Manipula múltiplas chamadas assíncronas via callback com Object.deferred (Você pode fragmentar esses recursos e adicionar no seu projeto, apenas o que for utilizar);
- Há uma vasta gama de Plugins a sua disposição;

Minha opinião ainda é **DEPENDE**, porém, acho que é um conhecimento válido, ele é fácil de ser aprendido, na verdade, o que precisar é só buscar na documentação, que é super fácil de ser compreendida.

**Reply by frontRocha** [reply]
JQuery é tudo, menos fácil de ler

**Comment by kvnol** [comment]
Issues iguais, @LFeh: #1080 #1142

**Comment by andrecoliveira** [comment]
Concordo com as palavras do @expalmer. Sempre é bom ter um pouco daquele conhecimento, por mais que não seja muito útil nos tempos de hoje. Mas um dia qualquer, irá ajudá-lo a resolver algum problema.

**Comment by matteusbarbosa** [comment]
Existe uma quantidade imensurável de plugins escritos em jQuery na web. Nem todo cógido com jQuery é ruim... Eu inclusive uso muitos deles no cotidiano, em projetos novos até. Me dá produtividade. Os plugins que mais uso: datatables, bootstrap, validationengine, jq-mask . Consigo fazer até uma simulação de componentes. Uso automatizador de tarefa como o gulp pra minificar as coisas e fico atento a código não utilizado em cada página... Cada página é auditada pra performance máxima. Não acho coerente condenar um plugin aperfeiçoado durante anos, quando sabemos que no final tudo vira javascript. Não dá pra simplesmente converter todos os projetos existentes p/ um novo framework, porque daqui uma hora surge outro novo framework da "hype". Me corrijam se estiver enganado! Obg
